src/common/transformations/src/transformations/common_optimizations/lora_subgraph_fusion.cpp

```diff
@@ -0,0 +1,108 @@
+
+#include <memory>
+#include <vector>
+
+#include "itt.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/convolution.hpp"
+#include "openvino/op/matmul.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/op/util/read_value_base.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/lora_subgraph.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::LoraSubgraphFusion::LoraSubgraphFusion() {
+    MATCHER_SCOPE(LoraSubgraphFusion);
+    using namespace pass::pattern;
+    auto lora_input_m = any_input();
+    auto transpose_const1_m = wrap_type<ov::op::v0::Constant>(consumers_count(1));
+    auto transpose1_m = optional<ov::op::v1::Transpose>({lora_input_m, transpose_const1_m}, consumers_count(1));
+    auto read_value1_m = wrap_type<ov::op::util::ReadValueBase>();
+    auto matmul1_m = wrap_type<ov::op::v0::MatMul>({transpose1_m, read_value1_m}, consumers_count(1));
+    auto read_value2_m = wrap_type<ov::op::util::ReadValueBase>();
+    auto multiply_m = wrap_type<ov::op::v1::Multiply>({matmul1_m, read_value2_m}, consumers_count(1));
+    auto read_value3_m = wrap_type<ov::op::util::ReadValueBase>();
+    auto matmul2_m = wrap_type<ov::op::v0::MatMul>({multiply_m, read_value3_m}, consumers_count(1));
+    auto transpose_const2_m = wrap_type<ov::op::v0::Constant>(consumers_count(1));
+    auto transpose2_m = optional<ov::op::v1::Transpose>({matmul2_m, transpose_const2_m}, consumers_count(1));
+    auto main_flow_m = wrap_type<ov::op::v0::MatMul, ov::op::v1::Convolution>({lora_input_m, any_input()});
+    auto add_m = wrap_type<ov::op::v1::Add>({transpose2_m, main_flow_m});
+
+    ov::matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto& lora_input = pattern_map.at(lora_input_m);
+        const auto& matmul1 = pattern_map.at(matmul1_m);
+        const auto& read_value1 = pattern_map.at(read_value1_m);
+        const auto& multiply = pattern_map.at(multiply_m);
+        const auto& read_value2 = pattern_map.at(read_value2_m);
+        const auto& matmul2 = pattern_map.at(matmul2_m);
+        const auto& read_value3 = pattern_map.at(read_value3_m);
+        const auto& main_flow = pattern_map.at(main_flow_m);
+        const auto& add = pattern_map.at(add_m);
+
+        const auto add_node = add.get_node_shared_ptr();
+        if (transformation_callback(add_node)) {
+            return false;
+        }
+
+        auto find_connected_input = [](ov::Node* child, ov::Node* parent) {
+            for (size_t i = 0; i < child->get_input_size(); ++i) {
+                auto input = child->input(i);
+                if (input.get_source_output().get_node() == parent)
+                    return input;
+            }
+            OPENVINO_THROW("Ops are not connected");
```

Comment:
> if pattern matcher works, I hope we will never visit this line.

Yes, that's right. And that's why I intentionally used `OPENVINO_THROW` here: an exception can be thrown only if the pattern was incorrectly changed, or there is a bug in the Matcher itself (strictly speaking, another possible reason is an incorrect usage of `find_connected_input`).
I thought that if this code path is unreachable in a healthy state of the product, we could throw an exception in order to simplify the debug in case of any issues (e.g. it's much easier to debug functional issue than performance issue, visible only in some specific cases, on large model).

I can correct the code in accordance to your suggestions, if you insist. Please let me know

Reply:
I agree, let's leave it as it is implemented.